pkg/graveler/staging/manager_test.go

```diff
@@ -7,19 +7,47 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/stretchr/testify/require"
 	"github.com/treeverse/lakefs/pkg/graveler"
 	"github.com/treeverse/lakefs/pkg/graveler/staging"
+	"github.com/treeverse/lakefs/pkg/kv"
+	kvparams "github.com/treeverse/lakefs/pkg/kv/params"
+	kvpg "github.com/treeverse/lakefs/pkg/kv/postgres"
 	"github.com/treeverse/lakefs/pkg/testutil"
 )
 
-func newTestStagingManager(t *testing.T) (context.Context, graveler.StagingManager) {
+func newTestStagingManager(t *testing.T, kvEnabled bool) (context.Context, graveler.StagingManager) {
 	t.Helper()
+	ctx := context.Background()
+	if kvEnabled {
+		// TODO (niro): Graveler unit tests running over postgres store until https://github.com/treeverse/lakeFS/issues/3622 is resolved
+		store, err := kv.Open(context.Background(), kvpg.DriverName, kvparams.KV{Postgres: &kvparams.Postgres{ConnectionString: databaseURI}})
+		testutil.MustDo(t, "Open KV Store", err)
+		kvStore := kv.StoreMessage{Store: store}
+		t.Cleanup(func() {
+			err = store.(*kvpg.Store).ClearTable(ctx)
+			if err != nil {
+				t.Fatalf("failed to delete KV table from postgres DB %s", err)
+			}
+			store.Close()
+		})
+		return ctx, staging.NewManager(kvStore)
+	}
 	conn, _ := testutil.GetDB(t, databaseURI)
-	return context.Background(), staging.NewManager(conn)
+	return context.Background(), staging.NewDBManager(conn)
 }
 
 func TestSetGet(t *testing.T) {
-	ctx, s := newTestStagingManager(t)
+	t.Run("TestDBSetGet", func(t *testing.T) {
```

Comment:
How about a general function that receives a testing function `func(t *testing.T, kvEnabled bool)`, and then put all the tests in array and iterate it from 2 function - one for `DB` and one for `KV`?
This will have a single point to delete later, when we abandon the `DB` support.
Take a look at [controller_test.go](https://github.com/treeverse/lakeFS/blob/master/pkg/api/controller_test.go)

Reply:
This was considered but it has a downside which in my opinion is worse. It is not possible to run single tests this way

Reply:
So even better - we can still have the general function `func actuallyRunTheTest` that receives the test func above and runs it once for DB and once for KV, and instead of putting all test functions in an array, leave the `func Test...` as is , but call the general testing `actuallyRunTheTest` providing it with the tests to run.
That way you can both run each `Test...` individually, and you minimized code to `runWithDB, runWithKV` to a single place
Makes sense?

Reply:
Not sure I completely follow but, take a look the changes and tell me if this is what you have in mind. Keep in mind we need the test names to be descriptive to differentiate between "DB" and "KV" tests